pennylane/gradients/vjp.py

```diff
@@ -37,6 +37,10 @@ def compute_vjp(dy, jac):
         return None
 
     dy_row = math.reshape(dy, [-1])
+
+    if not isinstance(dy_row, np.ndarray):
+        jac = math.convert_like(jac, dy_row)

```

Comment:
This is to take into account a case where `jac` was NumPy array, and `dy_row` was a tensor!